design-system/pkg/src/password-field/test/PasswordField.test.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { renderWithProvider } from '#test-utils';
+import userEvent from '@testing-library/user-event';
+
+import { PasswordField, PasswordFieldProps } from '..';
+
+let testId = 'test-id';
+let inputText = 'Hello world';
+
+function renderPasswordField(
+  props: Partial<PasswordFieldProps> = {},
+  ref?: React.RefObject<HTMLInputElement>
+) {
+  return renderWithProvider(
+    <PasswordField data-testid={testId} label="Password" {...props} ref={ref} />
+  );
+}
+
+describe('password-field/PasswordField', () => {
+  let onChangeSpy = jest.fn();
+  let user = userEvent.setup();
+
+  afterEach(() => {
+    onChangeSpy.mockClear();
+  });
+
+  it('renders basic elements correctly', () => {
+    let { getByLabelText } = renderPasswordField();
+    let input = getByLabelText('Password');
+
+    expect(input).toHaveAttribute('type', 'password');
+    expect(input).toHaveAttribute('autocomplete', 'current-password');
+  });
+
+  it('renders the reveal button', () => {
+    let { queryByLabelText } = renderPasswordField({
+      defaultValue: inputText,
+    });
+
+    expect(queryByLabelText('Show Password')).toBeTruthy();
```

Comment:
What do you think about using [toBeVisible](https://github.com/testing-library/jest-dom#tobevisible) to assert a few extra visibility traits?

```suggestion
    expect(getByLabelText('Show Password')).toBeVisible();
```

Reply:
Honestly wasn't familiar with this method. Seems more appropriate, thanks!